examples/plot_variable_importance_classif.py

```diff
@@ -0,0 +1,299 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy.linalg import toeplitz
+from scipy.stats import ttest_1samp
+from sklearn.base import clone
+from sklearn.linear_model import RidgeCV
+from sklearn.metrics import log_loss
+from sklearn.model_selection import RandomizedSearchCV, StratifiedKFold
+from sklearn.svm import SVC
+
+from hidimstat import CPI, PermutationImportance
+
+#############################################################################
+# Generate the data
+# ------------------------------
+# We generate the data using a multivariate normal distribution with a Toeplitz
+# correlation matrix. The target variable is generated using a non-linear function
+# of the features. To make the problem more intuitive, we generate a non-linear
+# combination of the features inspired by the Body Mass Index (BMI) formula.
+# The BMI can be obtained by :math:`\text{BMI} = \frac{\text{weight}}{\text{height}^2}`.
+# And we simply mimic the weight and height variables by rescalling 2 correlated
+# features. The binary target is then generated using the formula:
+# :math:`y = \beta_1 \exp\left(\frac{|\text{bmi} - \text{mean(bmi)}|}{\text{std(bmi)}}\right) + \beta_2 \exp\left(|\text{weight}| \times 1\left[|\text{weight} - \text{mean(weight)}| > \text{quantile(weight, 0.80)}\right] \right) + \beta_3 \cdot \text{age} + \epsilon` where :math:`\epsilon`` is a Gaussian noise.
+# The first and second term are non-linear functions of the features, corresponding to
+# deviations from the population mean while the third term is a linear function of a
+# feature.
+
+seed = 0
+rng = np.random.RandomState(seed)
+n_samples = 400
+n_features = 10
+corr_mat = toeplitz(
+    np.linspace(1, 0.0, n_features) ** 2,
+)
+mean = np.zeros(corr_mat.shape[0])
+X = rng.multivariate_normal(
+    mean,
+    corr_mat,
+    size=n_samples,
+)
+noise_y = 0.5
+weight = X[:, 4] * 20 + 74
+weight[weight < 40] += 30
+height = X[:, 2] * 0.12 + 1.7
+age = X[:, 6] * 10 + 50
+bmi = weight / height**2
+
+a = np.exp(np.abs(bmi - np.mean(bmi)) / np.std(bmi))
+b = np.exp(
+    np.abs(X[:, 4])
+    * ((X[:, 4] < np.quantile(X[:, 4], 0.10)) + (X[:, 4] > np.quantile(X[:, 4], 0.90)))
+)
+
+y_cont = 4 * a + 2.0 * b + 0.5 * age + noise_y * rng.randn(n_samples)
+y = y_cont > np.quantile(y_cont, 0.5)
+
+#############################################################################
+# Visualize the data
+# ------------------------------
+
+fig, axes = plt.subplots(
+    1,
+    2,
+    figsize=(8, 4),
+)
+axes[0].scatter(weight, height, c=y, cmap="coolwarm")
+axes[0].set_xlabel("Weight")
+axes[0].set_ylabel("Height")
+axes[1].matshow(
+    corr_mat,
+)
+labels = np.array(
+    [
+        "X1",
+        "X2",
+        "Height",
+        "X4",
+        "Weight",
+        "X6",
+        "Age",
+        "X8",
+        "X9",
+        "X10",
+    ]
+)
+tck_ids = np.arange(0, n_features)
+axes[1].set_xticks(tck_ids, labels[tck_ids], rotation=45)
+axes[1].set_yticks(tck_ids, labels[tck_ids])
+
```

Comment:
What do the red/blue dots correspond to ? Can you add a legend ?

Reply:
I updated the legend, it corresponds to variables for which the p-value is < threshold (here I used 0.05)